src/server/actions/user-actions.ts

```diff
@@ -40,3 +40,55 @@ export const findOrCreateSiUserId = async (clerkId: string, name: string) => {
 
     return user.id
 }
+
+export const createUserAction = async (
+    clerkId: string,
+    name: string,
+    isResearcher: boolean,
+    memberId: string,
+    isReviewer: boolean,
+) => {
+    // Run all related inserts in a transaction for atomicity
+    return await db.transaction().execute(async (trx) => {
+        // Check if a user with the given clerkId already exists
+        const existingUser = await trx.selectFrom('user').select('id').where('clerkId', '=', clerkId).executeTakeFirst()
+
+        if (existingUser) {
+            throw new Error(`User with clerkId ${clerkId} already exists`)
+        }
+
+        // Insert the user record
+        const results = await trx
+            .insertInto('user')
+            .values({
+                clerkId: clerkId,
+                name,
+                isResearcher: isResearcher,
+            })
+            .returningAll()
+            .execute()
+
+        if (!results.length) {
+            throw new Error('Failed to create user in DB')
+        }
+
+        // Insert into member_user table using provided memberId and isReviewer flags.
+        const memberUserResults = await trx
+            .insertInto('memberUser')
+            .values({
+                userId: results[0].id,
+                memberId: memberId,
+                isReviewer: isReviewer,
+                isAdmin: false,
+            })
+            .returningAll()
+            .execute()
+
+        if (!memberUserResults.length) {
+            throw new Error('Failed to create member_user record')
+        }
+
+        // Return the created user record.
+        return results[0]
```

Comment:
Rather than getting an array back from the query (using `.execute()`) you can use `.executeTakeFirst()` or `.executeTakeFirstOrThrow()` if you want it to throw on error